qiskit/converters/dag_to_circuit.py

```diff
@@ -75,6 +76,19 @@ def dag_to_circuit(dag, copy_operations=True):
     circuit.metadata = dag.metadata or {}
     circuit._data = circuit_data
 
-    circuit._duration = dag.duration
-    circuit._unit = dag.unit
+    with warnings.catch_warnings():
```

Comment:
Instead of filtering the warning why not just use the internal attribute which doesn't have the warning?